### PR TITLE
[Built-in rule] Add `SystemAssigned, UserAssigned` as an allowed identity on TA-000007, TA-000013, TA-000019

### PR DESCRIPTION
**Pull request by richardsondev**
## Description

Add `SystemAssigned, UserAssigned` as an accepted identity on rules TA-000007, TA-000013, TA-000019. This is a valid operation on Azure Functions and currently our bicep templates are being flagged by this so we need to ignore this rule.

Issue #372

### Expected behavior

TA-000013 should not detect Azure Functions as missing an identity when both SystemAssigned and UserAssigned are defined:

```bicep
  identity: {
    type: 'SystemAssigned, UserAssigned'
    userAssignedIdentities: {
      '${managedIdentity.id}': {}
    }
  }
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] The pull request does not introduce breaking changes.

### [General Guidelines](../CONTRIBUTING.md)
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request is clear and informative.
- [ ] I have added myself to the 'assignees'. (no permissions)
- [x] I have added 'linked issues' if relevant.

### [Testing Guidelines](../CONTRIBUTING.md#tests)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/Analyzer.Core/Rules/BuiltInRules.json [hunk 1]
```diff
@@ -394,7 +394,8 @@
           "path": "identity.type",
           "in": [
             "UserAssigned",
-            "SystemAssigned"
+            "SystemAssigned",
+            "SystemAssigned, UserAssigned"
```
Comment (VeraBE) [inline src/Analyzer.Core/Rules/BuiltInRules.json:398]: Reminder to consider the template might say "SystemAssigned,UserAssigned" (no space)

Comment (richardsondev) [reply src/Analyzer.Core/Rules/BuiltInRules.json]: Good catch, just added support and tests for this.

As a side-note: I did see that the ARM/bicep has a warning when the space is missing `Value must be one of the following values: "SystemAssigned", "UserAssigned", "SystemAssigned, UserAssigned", "None"`

**Review by reynoldsa — APPROVED** [review]
Looks good

File: src/Analyzer.Core/Rules/BuiltInRules.json [hunk 2]
```diff
@@ -392,10 +392,7 @@
       "allOf": [
         {
           "path": "identity.type",
-          "in": [
-            "UserAssigned",
-            "SystemAssigned"
-          ]
+          "regex": "^(UserAssigned|SystemAssigned|SystemAssigned, *UserAssigned)$"
```
Comment (JohnathonMohr) [inline src/Analyzer.Core/Rules/BuiltInRules.json:395]: The [documentation for this property](https://learn.microsoft.com/en-us/azure/templates/microsoft.web/sites?pivots=deployment-language-arm-template#managedserviceidentity-1) uses exactly one space, and doesn't specify any other details, so I would actually err on the side of caution and only allow for exactly one space, as the doc states. You can then change this operator back to `in` for better readability.

File: src/Analyzer.Core/Rules/BuiltInRules.json [hunk 3]
```diff
@@ -392,10 +392,7 @@
       "allOf": [
```
Comment (JohnathonMohr) [inline src/Analyzer.Core/Rules/BuiltInRules.json:392]: Can you also please update rules TA-000007 and TA-000019? They check the same thing, but for Web Apps and API Apps.

**Review by JohnathonMohr — CHANGES_REQUESTED** [review]
Thank you for your contribution! Just a couple comments.

File: src/Analyzer.Core/Rules/BuiltInRules.json (hunk 1 above)
Comment (richardsondev) [reply src/Analyzer.Core/Rules/BuiltInRules.json]: Based on the recent feedback, I will revert this change to only allow a single space.

File: src/Analyzer.Core/Rules/BuiltInRules.json (hunk 2 above)
Comment (richardsondev) [reply src/Analyzer.Core/Rules/BuiltInRules.json:395]: Sounds good, that will align with the Bicep build warning as well when the space is missing.

File: src/Analyzer.Core/Rules/BuiltInRules.json (hunk 3 above)
Comment (richardsondev) [reply src/Analyzer.Core/Rules/BuiltInRules.json]: Updated these rules and added some tests

File: docs/built-in-rules.md [hunk 4]
```diff
@@ -83,7 +83,7 @@ Cross-Origin Resource Sharing (CORS) should not allow all domains to access your
 ### TA-000013: Managed identity should be used in your function app
 For enhanced authentication security, use a managed identity. On Azure, managed identities eliminate the need for developers to have to manage credentials by providing an identity for the Azure resource in Azure AD and using it to obtain Azure Active Directory (Azure AD) tokens.
 
-**Recommendation**: To [use Managed Identity](https://docs.microsoft.com/azure/app-service/overview-managed-identity?tabs=dotnet), in the [Microsoft.Web/sites resource managed identity property](https://docs.microsoft.com/azure/templates/microsoft.web/sites?tabs=json#ManagedServiceIdentity), add (or update) the *type* property, setting its value to `"SystemAssigned"` or `"UserAssigned"` and providing any necessary identifiers for the identity if required.
```
Comment (richardsondev) [inline docs/built-in-rules.md:86]: Update docs for other rules

**Review by JohnathonMohr — APPROVED** [review]

**Pull request merged**